python/taichi/lang/matrix.py

```diff
@@ -1614,7 +1614,7 @@ def cast(self, mat):
         return mat.cast(self.dtype)
 
     def filled_with_scalar(self, value):
-        return Matrix([[value for _ in range(self.m)] for _ in range(self.n)])
+        return Matrix([[ops_mod.cast(value, self.dtype) for _ in range(self.m)] for _ in range(self.n)])
```

Comment:
without this explicit type casting, the matrix will have an inferred type, so
```
v = vec3(1)
```
will have integral entries `[1, 1, 1]`.